packages/cozy-doctypes/src/BankTransaction.js

```diff
@@ -99,14 +104,30 @@ class Transaction extends Document {
 
       if (!existingTransactions) {
         missedTransactions.push(...newTransactions)
+
+        if (typeof options.onCatchupNonExistentTransaction === 'function') {
+          options.onCatchupNonExistentTransaction()
+        }
+
         continue
       }
 
       if (newTransactions.length > existingTransactions.length) {
         const difference = newTransactions.length - existingTransactions.length
         missedTransactions.push(...newTransactions.slice(0, difference))
+
+        if (typeof options.onCatchupMoreNew === 'function') {
+          options.onCatchupMoreNew()
+        }
+
         continue
       }
+
+      if (newTransactions.length < existingTransactions.length) {
+        if (typeof options.onCatchupLessNew === 'function') {
+          options.onCatchupLessNew()
+        }
+      }
```

Comment:
I dont' find the names of the functions very telling :/ 

Shouldn't `onCatchupLessNew` and `onCatchupMoreNew` have `Transaction` also ?

* Maybe the callbacks could take the current identifier ?

Catchup is not the right verb I think (Catch up == rattraper son retard): `onFetchTransaction(newTransactions, existingTransactions)` and then it is the responsibility of the caller to discriminate and send the right piwik events

Reply:
Cool :) `onMissedTransactionFound` seems clearer too :)